File: app/src/main/java/com/haker/ah/CloudAdapter.java
package com.haker.ah;


import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

/**
 * Created by js on 2018-05-28.
 */

public class CloudAdapter extends RecyclerView.Adapter<CloudAdapter.MyViewHolder>{
    private ArrayList<CloudData> mDataset;

    public class MyViewHolder extends RecyclerView.ViewHolder {
        public TextView name, lan, lon;

        //ViewHolder
        public MyViewHolder(View view) {
            super(view);
            name = (TextView) view.findViewById(R.id.name);
        }
    }

    public CloudAdapter(ArrayList<CloudData> myData){
        this.mDataset = myData;
    }

    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.cloud_item, parent, false);
        return new MyViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull CloudAdapter.MyViewHolder holder, int position) {

        holder.name.setText(mDataset.get(position).getName());


    }

    @Override
    public int getItemCount() {
        return mDataset.size();
    }
}

File: app/src/main/java/com/example/alt_map/CloudAdapter.kt
package com.example.alt_map

import android.content.Context
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

class CloudAdapter(val context: Context) : RecyclerView.Adapter<CloudAdapter.Holder>() {

    var data = mutableListOf<CloudData>()

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): Holder {
        Log.d("상태","CloudAdapter_onCreateViewHolder")
        val view = LayoutInflater.from(context).inflate(R.layout.cloud_item, parent, false)
        return Holder(view)
    }

    override fun onBindViewHolder(holder: Holder, position: Int) {
        Log.d("상태","CloudAdapter_onBindViewHolder")
        val cloudData = data[position]
        holder.bind(cloudData)
    }

    override fun getItemCount(): Int {
        Log.d("상태","CloudAdapter_getItemCount")
        return data.size
    }

    inner class Holder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val name = itemView.findViewById<TextView>(R.id.cloud_activity_text)
        val img = itemView.findViewById<ImageView>(R.id.profile_img)
        fun bind(cloudData: CloudData) {
            name.text = cloudData.name
            img.setImageResource(cloudData.profile)
            img.clipToOutline = true
        }
    }
}